codecov_cli/parsers/base.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime, timedelta
+from typing import List
+
+
+class ParsingError(Exception):
+    ...
+
+
+class Test(object):
+    def __init__(self, name: str, status: bool, duration: timedelta):
+        self.name = name
+        self.status = status
+        self.duration = duration
+
+    def __repr__(self):
+        return f"{self.name}:{self.status}:{self.duration.total_seconds()}::"
+
+
+class Testrun(object):
+    def __init__(
+        self,
+        name: str,
+        timestamp: datetime,
+        time: timedelta,
+        testcases: List[Test],
+        failures: int,
+        errors: int,
+        skipped: int,
+        total: int,
+    ):
+        self.testcases = testcases or []
```

Comment:
If you don't specify a default value for test cases (e.g. `testcases: List[Test] = None` then it can't ever be `None`, right?

(if it can then the type hint should be `Optional`)